src/main.rs

```diff
@@ -813,6 +853,157 @@ async fn handle_request(
     }
 }
 
+fn setup_http_client(
+    logger: &slog::Logger,
+    danger_accept_invalid_certs: bool,
+    root_certificates: &[PathBuf],
+    addr_mappings: Vec<OptResolve>,
+) -> reqwest::Client {
+    let builder = rustls::ClientConfig::builder().with_safe_defaults();
+    let mut tls_config = if !danger_accept_invalid_certs {
+        use rustls::Certificate;
+        use rustls::RootCertStore;
+
+        let mut root_cert_store = RootCertStore::empty();
+        for cert_path in root_certificates {
+            let mut buf = Vec::new();
+            if let Err(e) = File::open(cert_path).and_then(|mut v| v.read_to_end(&mut buf)) {
+                slog::warn!(
+                    logger,
+                    "Could not load cert `{}`: {}",
+                    cert_path.display(),
+                    e
+                );
+                continue;
+            }
+            match cert_path.extension() {
+                Some(v) if v == "pem" => {
```

Comment:
Would it make sense to just require that the cert be in a given format (either `pem` or `der`)?